MapboxDirections/MBDirectionOptions.swift

```diff
@@ -184,8 +184,11 @@ public enum InstructionFormat: UInt, CustomStringConvertible {
     }
 }
 
-@objc(MBDirecitonOptions)
-open class DirectionOptions: NSObject, NSSecureCoding, NSCopying {
+/**
+ A base class that contains shared resources between `MatchOptions` and `RouteOptions`.
```

Comment:
This is a public class, so document what the developer needs to know, not what we need to know internally:

> Options for calculating results from the Mapbox Directions service.
> 
> You do not create instances of this class directly. Instead, create instances of `MatchOptions` or `RouteOptions`.